ctapipe/image/extractor.py

```diff
@@ -1051,7 +1052,9 @@ def _apply_second_pass(
         # Apply correction to 1st pass charges
         charge_1stpass = charge_1stpass_uncorrected * correction[selected_gain_channel]
 
-        camera_geometry = self.subarray.tel[tel_id].camera.geometry
+        camera_geometry = self.subarray.tel[tel_id].camera.geometry.transform_to(
+            TelescopeFrame()
```

Comment:
this might need some thought - we shouldn't be transforming the geometry for each event *inside* a calibration/image-extraction algorithm like this as that adds quite a bit of overhead.   This algorithm does not care about the frame of the camera geometry or pointing corrections, so it should not require a transform. 

Since we do eventually need to transform for the ImageProcessor, it may be better to do the transform outside of these algorithms, perhaps in the CameraCalibrator (which eventually should also apply pointing correcions), so that a single TelescopeFrame is available to all algorithms that need it.  That would avoid re-doing the transform many times.

Reply:
Yes, this should be done inside the `ProcessorTool` in `setup` after writing out the subarray description but before setting up telescope components. The telescope components should just get a subarray with cameras in `CameraFrame`.

Reply:
Maybe add `transform_camera_geometries` to `SubarrayDescription` returning a new Subarray with transformed geometries

Reply:
Is creating a whole new SubarrayDescription each time efficient? This was an issue I think we discussed a while back (need to find where): during analysis, the TelescopeFrame needs to be updated once per event for observations, and could be simplified to only once per run for  simulations in fixed Alt/Az mode.

The TelescopeFrame requires: 
* the latest pointing position, which changes in time in ICRS tracking mode (the default for most observations)
* pre-computed pointing corrections which could include not just a shift in RA/Dec, but also a rotation (this is not currently implemented, but will need to be done eventually)

Reply:
Ah, sorry, forgot about pointing corrections.

But it stands: we could also do the pointing corrections in the transformation from / to TelescopeFrame to higher frames. That would actually make the most sense to me, since pointing direction is not important for the `CameraFrame` <-> `TelescopeFrame` transformation.

To me that means we should transform the whole subarray *once* to `TelescopeFrame` and apply corrections in the transformation `AltAz` <-> `TelescopeFrame`, we don't need to change telescope frame for this, as the attributes needed for the transformation can be specified on the `SkyCoord` as seen here:

https://github.com/cta-observatory/ctapipe/blob/8a7a367dfe4a2f4e9bdf0e9bf774be29bf03b75f/ctapipe/visualization/mpl_camera.py#L434-L439

Reply:
Remember there are in principle rotation corrections and maybe others, so you have to apply pointing corrections _before_ Hillas parameterization (doing it while going to Alt/Az wouldn't work), as you need to have correctly shifted/rotated parameters before you do shower geometry reconstruction.  But that can be solved when the ShowerProcessor is moved to TelescopeFrame later (right now e.g. HillasReconstructor supports both frames, but that's ugly).